src/module/actor/mixins/actor-conditions.js

```diff
@@ -21,21 +33,38 @@ export const ActorConditionsMixin = (superclass) => class extends superclass {
         return conditionItem;
     }
 
-    async setCondition(conditionName, enabled) {
+    /**
+     * Updates the Actor's conditions. Either adds or removes a condition Item as necessary to match the enabled argument.
+     * @param {String} conditionName The name of the condition. Must match any key from config.js SFRPG.statusEffectIconMapping. Case insensitive.
+     * @param {Boolean} enabled If this value is true it ensures the condition is present on the Actor.
+     * @param {Object} overlay If this value is true it indicates that the token icon should be added as a full-sized overlay. Default is false.
+     * @returns {Promise<*>} The Promise resulting from the create or delete Embedded Document call.
+     */
+    async setCondition(conditionName, enabled, { overlay = false } = {}) {
         if (!SFRPG.statusEffectIconMapping[conditionName]) {
             ui.notifications.warn(`Trying to set condition ${conditionName} on actor ${this.name} but the condition is not valid. See CONFIG.SFRPG.statusEffectIconMapping for all valid conditions.`);
             return;
         }
 
-        const conditionItem = this.getCondition(conditionName);
+        // Try to get status effect object as a workaround for a poorly conceived check in foundry.js Token.toggleEffect(...)
```

Comment:
Status effect for loop can be replaced with something like:
let statusEffect = SFPRG.statusEffects.find(x => x.id === conditionName);